pkg/libvmi/status/status.go

```diff
@@ -0,0 +1,123 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors
+ *
+ */
+
+// Package libvmistatus provides utility options to manage the vmi status.
+// This package should be used only in unit test files in which there is the need
+// to set the status of a vmi to simulate different behaviors.
+// BE AWARE: any usage of this package outside the unit test files is to be considered wrong
+// since the status should only be manipulated by the controllers.
+package libvmistatus
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/libvmi"
+)
+
+type StatusOption func(vmiStatus *v1.VirtualMachineInstanceStatus)
+
+// WithStatus sets the status with specified value
+func WithStatus(status v1.VirtualMachineInstanceStatus) libvmi.Option {
+	return func(vmi *v1.VirtualMachineInstance) {
+		vmi.Status = status
+	}
+}
+
+// New instantiates a new VMI status configuration,
+// building its properties based on the specified With* options.
+func New(opts ...StatusOption) v1.VirtualMachineInstanceStatus {
+	vmiStatus := &v1.VirtualMachineInstanceStatus{}
+	for _, f := range opts {
+		f(vmiStatus)
+	}
+
+	return *vmiStatus
+}
+
+func WithBaseStatus(status *v1.VirtualMachineInstanceStatus) StatusOption {
+	return func(vmiStatus *v1.VirtualMachineInstanceStatus) {
+		*vmiStatus = *status
+	}
+}
+
+func Update(status v1.VirtualMachineInstanceStatus, opts ...StatusOption) v1.VirtualMachineInstanceStatus {
```

Comment:
Perhaps you should document this function.

You should also warn the user that the inputted argument may be mutated (e.g. its lists or content at the pointers). It is up to the caller to pass a copy (using `DeepCopy()`) if protecting the data is needed.